### ISSUE TITLE
Implement SobolSampler

### ISSUE THREAD

**Issue by wahn** [wahn/rs_pbrt#25]
The example scene **assets/scenes/anim-bluespheres.pbrt** uses a **SobolSampler**:

```shell
> grep Sampler assets/scenes/anim-bluespheres.pbrt 
Sampler "sobol" "integer pixelsamples" [1024]
```

Right now we have only one implementor of the **trait Sampler**:

```shell
> grep "impl Sampler for" src/lib.rs
impl Sampler for ZeroTwoSequenceSampler {
```

The C++ side has two **global** samplers:

```shell
> rg -tcpp ": public GlobalSampler" ~/git/github/pbrt-v3/src/
/home/jan/git/github/pbrt-v3/src/samplers/sobol.h
47:class SobolSampler : public GlobalSampler {

/home/jan/git/github/pbrt-v3/src/samplers/halton.h
48:class HaltonSampler : public GlobalSampler {
```

This ticket has the task to implement the **class SobolSampler**, issue #9 is about implementing the **class HaltonSampler** ...

**Comment by wahn** [comment]
This is how the result should look like for the **Cornell Box** (rendered by the **C++ version**):

![pbrt_cpp](https://user-images.githubusercontent.com/1074865/33128274-b5bc5ef2-cf83-11e7-81b0-5c699230cdcf.png)

```diff
diff --git a/pbrt/assets/scenes/cornell_box.pbrt b/pbrt/assets/scenes/cornell_box.pbrt
index 3838487..aa3a210 100644
--- a/pbrt/assets/scenes/cornell_box.pbrt
+++ b/pbrt/assets/scenes/cornell_box.pbrt
@@ -10,7 +10,7 @@ Film "image"
   "integer yresolution" [ 500 ]
 ##  "integer outlierrejection_k" [ 10 ]
 ##Sampler "sobol"
-Sampler "halton" "integer pixelsamples" [8]
+Sampler "sobol" "integer pixelsamples" [8]
 ##PixelFilter "blackmanharris"
 ##SurfaceIntegrator "bidirectional"
 ##Integrator "directlighting" "integer maxdepth" [10]
```

**Comment by wahn** [comment]
This is the same scene rendered by commit 1b4dd8dbb0e90ed3890bb82d5adab374c08dd77e and the resulting executable:

![pbrt](https://user-images.githubusercontent.com/1074865/33128588-cf1874d4-cf84-11e7-80b9-9f85c1e14b55.png)

Let's find out what's going wrong here ...

**Comment by wahn** [comment]
The last commit 5589efcb6343c02b9b870e60d53681288ddc8189 resolves issue #25:

![pbrt](https://user-images.githubusercontent.com/1074865/33137886-0191cbba-cfa1-11e7-8b3c-806028b15666.png)

```shell
> imf_diff pbrt.exr pbrt_rust.exr
pbrt.exr pbrt_rust.exr: no differences.
== "pbrt.exr" and "pbrt_rust.exr" are identical
```
